sdk/synapse/synapse-artifacts/src/operations/dataFlowDebugSession.ts

```diff
@@ -89,71 +101,117 @@ export class DataFlowDebugSessionImpl implements DataFlowDebugSession {
    * @param request Data flow debug session definition
    * @param options The options parameters.
    */
-  async createDataFlowDebugSession(
+  async beginCreateDataFlowDebugSession(
     request: CreateDataFlowDebugSessionRequest,
-    options?: coreHttp.OperationOptions
-  ): Promise<LROPoller<DataFlowDebugSessionCreateDataFlowDebugSessionResponse>> {
-    const { span, updatedOptions } = createSpan(
-      "ArtifactsClient-createDataFlowDebugSession",
+    options?: DataFlowDebugSessionCreateDataFlowDebugSessionOptionalParams
+  ): Promise<
+    PollerLike<
+      PollOperationState<
+        DataFlowDebugSessionCreateDataFlowDebugSessionResponse
+      >,
+      DataFlowDebugSessionCreateDataFlowDebugSessionResponse
+    >
+  > {
+    const { span } = createSpan(
+      "ArtifactsClient-beginCreateDataFlowDebugSession",
       options || {}
     );
-    const operationArguments: coreHttp.OperationArguments = {
-      request,
-      options: this.getOperationOptions(updatedOptions, "undefined")
-    };
-    const sendOperation = async (
-      args: coreHttp.OperationArguments,
-      spec: coreHttp.OperationSpec
-    ) => {
+    const directSendOperation = async (
+      args: coreClient.OperationArguments,
+      spec: coreClient.OperationSpec
+    ): Promise<DataFlowDebugSessionCreateDataFlowDebugSessionResponse> => {
       try {
         const result = await this.client.sendOperationRequest(args, spec);
         return result as DataFlowDebugSessionCreateDataFlowDebugSessionResponse;
       } catch (error) {
         span.setStatus({
-          code: SpanStatusCode.ERROR,
+          code: coreTracing.SpanStatusCode.UNSET,
           message: error.message
         });
         throw error;
       } finally {
         span.end();
       }
     };
+    const sendOperation = async (
```

Comment:
just venting: not happy with the boilerplate duplication here by code gen, here is the tracking issue: https://github.com/Azure/autorest.typescript/issues/1079

happy to see `LroEngine` used in more packages though :)